src/contracts/core/DelegationManager.sol

```diff
@@ -224,34 +233,40 @@ contract DelegationManager is Initializable, OwnableUpgradeable, Pausable, Deleg
             "DelegationManager.undelegate: caller cannot undelegate staker"
         );
 
-        // remove any shares from the delegation system that the staker currently has delegated, if necessary
-        // force the staker into "undelegation limbo" in the EigenPodManager if necessary
-        if (eigenPodManager.podOwnerHasActiveShares(staker)) {
-            uint256 podShares = eigenPodManager.forceIntoUndelegationLimbo(staker, operator);
-            // remove delegated shares from the operator
-            _decreaseOperatorShares({
-                operator: operator,
-                staker: staker,
-                strategy: beaconChainETHStrategy,
-                shares: podShares
-            });
-        }
-        // force-queue a withdrawal of all of the staker's shares from the StrategyManager, if necessary
-        if (strategyManager.stakerStrategyListLength(staker) != 0) {
-            IStrategy[] memory strategies;
-            uint256[] memory strategyShares;
-            (strategies, strategyShares, withdrawalRoot) = strategyManager.forceTotalWithdrawal(staker);
```

Comment:
where did `forceTotalWithdrawal` go?

Reply:
With this change, we no longer have a need for `StrategyManager.forceTotalWithdrawal` or `EigenPodManager.forceIntoUndelegationLimbo` -- delegation limbo is more "contained" just in the DelegationManager contract itself.
The flip side of this is that if a staker is in delegation limbo, we currently:

- prevent them from queuing a new withdrawal (in either the StrategyManager or DelegationManager)
- prevent them from re-delegating

I think however that we could allow queuing a withdrawal when in delegation limbo? Would just need to store the proper operator address for the queued withdrawal.
We could also then allow exiting delegation limbo early if a user subsequently removes all their shares via withdrawals.
I feel like these features could improve UX, but it's kind of weird because if implemented naively it might "restart the clock" on the withdrawal delay for the funds, so if we decide to add this we need to be careful about implementation.

Reply:
got it, thanks :)